microproxy/proxy.py

```diff
@@ -86,28 +87,28 @@ def next_layer(self, current_layer, context):
 
 
 class ProxyServer(tcpserver.TCPServer):
-    def __init__(self, config, proxy_server_handler=None):
-        super(ProxyServer, self).__init__()
+    def __init__(self, config, **kwargs):
+        super(ProxyServer, self).__init__(**kwargs)
         self.config = config
-        self.host = config["host"]
-        self.port = config["port"]
         self.interceptor = Interceptor(config=config)
+        self.layer_manager = LayerManager(config)
```

Comment:
Sorry, it's my bad that I didn't notice that you move layer_manager into handle_stream.
But I think there is no reason to create each instance of LayerManager for each connection,
because LayerManager is stateless and thread-safe.

And if we remove LayerManager from ProxyServer's field, it would turn into a hard to test pattern that we could not mock LayerManager if we want.